tests/test_m2m.py

```diff
@@ -0,0 +1,159 @@
+
+    def testMapUnqualifiedAgentNoOptions(self):
+        self.record = m2m.MetadataRecord("mphillips")
+        self.record.map("agent", "creator", "Phillips, Mark")
+
+        self.assertEquals(self.record.__str__(), '<?xml version="1.0" encoding="UTF-8"?>\n<metadata>\n  <creator>\n    <name>Phillips, Mark</name>\n  </creator>\n  <meta qualifier="metadataCreator">mphillips</meta>\n</metadata>\n')
+
+    def testMapQualifiedAgentNoOptions(self):
+        self.record = m2m.MetadataRecord("mphillips")
+        self.record.map("agent", "creator", "Phillips, Mark", qualifier="aut")
+
+        self.assertEquals(self.record.__str__(), '<?xml version="1.0" encoding="UTF-8"?>\n<metadata>\n  <creator qualifier="aut">\n    <name>Phillips, Mark</name>\n  </creator>\n  <meta qualifier="metadataCreator">mphillips</meta>\n</metadata>\n')
+
+    def testMapQualifiedAgentInfoOnly(self):
+        self.record = m2m.MetadataRecord("mphillips")
+        self.record.map("agent", "creator", "Phillips, Mark", 
+                        qualifier="aut", info="First Publication")
+
+        self.assertEquals(self.record.__str__(), '<?xml version="1.0" encoding="UTF-8"?>\n<metadata>\n  <creator qualifier="aut">\n    <info>First Publication</info>\n    <name>Phillips, Mark</name>\n  </creator>\n  <meta qualifier="metadataCreator">mphillips</meta>\n</metadata>\n')
+
+    def testMapQualifiedAgentInfoAndAgentType(self):
+        self.record = m2m.MetadataRecord("mphillips")
+        self.record.map("agent", "creator", "Phillips, Mark", 
+                        qualifier="aut",
+                        info="First Publication",
+                        agent_type="per")
+
+        self.assertEquals(self.record.__str__(), '<?xml version="1.0" encoding="UTF-8"?>\n<metadata>\n  <creator qualifier="aut">\n    <info>First Publication</info>\n    <type>per</type>\n    <name>Phillips, Mark</name>\n  </creator>\n  <meta qualifier="metadataCreator">mphillips</meta>\n</metadata>\n')
+
+    def testMapQualifiedAgentWithLocation(self):
+        self.record = m2m.MetadataRecord("mphillips")
+        self.record.map("agent", "publisher", "UNT Libraries", 
+                        location="Denton, Texas")
+
+        self.assertEquals(self.record.__str__(), '<?xml version="1.0" encoding="UTF-8"?>\n<metadata>\n  <publisher>\n    <name>UNT Libraries</name>\n    <location>Denton, Texas</location>\n  </publisher>\n  <meta qualifier="metadataCreator">mphillips</meta>\n</metadata>\n')
+
+    def testIncorrectUseOfLocationInAgentThatIsNotPublisher(self):
+
+        with self.assertRaises(m2m.MetadataConverterException) as cm:
+            self.record = m2m.MetadataRecord("mphillips")
+            self.record.map("agent", "creator", "Phillips, Mark", 
+                        location="aut")
```

Comment:
i don't like to leave PEP comments, but ...

This line is under indented. :smile: